service/src/sync_state.rs

```diff
@@ -0,0 +1,92 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Handles all state syncing mission of a worker before start up.
```

Comment:
This file was introduced as a preparatory step for requesting the state of a fellow validateer (requesting keys will be changed to request keys & state). Might be moved to the sidechain crate later on. But for this PR I think it's fine like that.